### PR TITLE
Update mmdetection patch integrating NNCF

### PR DESCRIPTION
**Pull request by aleksei-kashapov**
Add new config files
Integrate NNCF support into test.py
Add export to ONNX format ability for NNCF-compressed model
Improve code style

### CODE REVIEW

**Comment by vshampor** [comment]
@aleksei-kashapov please adjust the third-party integration sanity tests as well:
https://github.com/openvinotoolkit/nncf_pytorch/blob/develop/tests/test_sanity_third_party.py

**Comment by vanyalzr** [comment]
Why rename the patch? In any case, we should think about renaming the transformers patch accordingly to be consistent.

File: third_party_integration/mmdetection/0001-MMDetection-patch-integrating-NNCF-support.patch [hunk 1]
```diff
@@ -0,0 +1,1439 @@
...
++
++        cfg.nncf_load_from = args.checkpoint
++        compression_ctrl, model = wrap_nncf_model(model, cfg, None)
++
++        input_size = input_shape
++        device = "cpu"
++        input_args = ([torch.randn(input_size).to(device), ],)
++        input_kwargs = dict(return_loss=False, dummy_forward=True)
++        compression_ctrl.export_model(args.output_file, *input_args, **input_kwargs)
++        print ("Successfully convert compressed model to ONNX format")
```
Comment (vanyalzr) [inline third_party_integration/mmdetection/0001-MMDetection-patch-integrating-NNCF-support.patch:1281]: converted

**Comment by aleksei-kashapov** [comment]
Make a new pull request

**Pull request closed without merging**